client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
 		RecordType:   recordType,
 	})
 }
+
+type Secret struct {
+	SecretType    string
+	SecretName    string
+	SecretValue   string
+	Description   string
+	FileName      string
+	SecretID      string
+	RealmName     string
+	OwnerUsername string
+	NamespaceId   string
+	Version       string
+	Record        *pdsClient.Record
+}
+
+type CreateSecretOptions struct {
+	SecretType  string
+	SecretName  string
+	SecretValue string
+	Description string
+	FileName    string
+	RealmName   string
+}
+
+// CreateSecret makes a secret of the specified type and share it with a group containing the writer
+func (c *ToznySDKV3) CreateSecret(ctx context.Context, secret CreateSecretOptions) (*Secret, error) {
+	// once ToznySDKV3 is updated, this will be c.RealmName is secret.RealmName is empty
+	realmName := secret.RealmName
+	err := ValidateSecret(secret)
+	if err != nil {
+		return nil, err
+	}
+	group, err := c.GetOrCreateNamespace(ctx, realmName, c.StorageClient.ClientID)
+	if err != nil {
+		return nil, err
+	}
+	recordType := fmt.Sprintf("tozny.secret.%s.%s.%s", SECRET_UUID, secret.SecretType, secret.SecretName)
+	timestamp := time.Now().String()
+	plain := map[string]string{
+		"secretType":  secret.SecretType,
+		"secretName":  secret.SecretName,
+		"description": secret.Description,
+		"version":     timestamp,
+	}
+	// var createdSecret *Secret
+	var createdRecord *pdsClient.Record
+	if secret.SecretType == "File" {
+		createdRecord, err = c.WriteFile(ctx, recordType, plain, secret.FileName)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		data := map[string]string{"secretValue": secret.SecretValue}
+		createdRecord, err = c.WriteRecord(ctx, data, recordType, plain)
+		if err != nil {
+			return nil, err
+		}
+	}
+	createdSecret := c.MakeSecretResponse(createdRecord, group.GroupID.String(), "")
+	err = c.ShareRecordWithGroup(ctx, recordType, group)
+	if err != nil {
+		return nil, err
+	}
+	return createdSecret, nil
+}
+
+// WriteFile encrypts the file with specified fileName and uploads it, creating a new record in E3DB
+func (c *ToznySDKV3) WriteFile(ctx context.Context, recordType string, plain map[string]string, fileName string) (*pdsClient.Record, error) {
+	keyReq := pdsClient.GetOrCreateAccessKeyRequest{
+		WriterID:   c.E3dbPDSClient.ClientID,
+		UserID:     c.E3dbPDSClient.ClientID,
+		ReaderID:   c.E3dbPDSClient.ClientID,
+		RecordType: recordType,
+	}
+	ak, err := c.E3dbPDSClient.GetOrCreateAccessKey(ctx, keyReq)
+	if err != nil {
+		return nil, err
+	}
+	// Encrypt the file
+	size, checksum, err := e3dbClients.EncryptFile(fileName, encryptedFileName, ak)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := os.Remove(encryptedFileName)
+		if err != nil {
+			fmt.Println("CreateSecret: error removing encrypted")
```

Comment:
imagine you run this command and get this error. Now imagine you didn't write this code. What would you think this message means, and how could you make it more obvious?